src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CultureInfoAdapter.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Runtime.Serialization
+{
+    [DataContract(Name = "CultureInfo", Namespace = "http://schemas.datacontract.org/2004/07/System.Globalization")]
+    public struct CultureInfoAdapter
+    {
+        private string _name;
+        private int _dataItem;
+        private bool _isReadOnly;
+        private bool _useUserOverride;
+
+        [DataMember(Name = "calendar", IsRequired = true)]
+        public string Calendar { get; set; }
+
+        [DataMember(Name = "compareInfo", IsRequired = true)]
+        public string CompareInfo { get; set; }
+
+        [DataMember(Name = "cultureID", IsRequired = true)]
+        public int CultureID { get; set; }
+
+        [DataMember(Name = "dateTimeInfo", IsRequired = true)]
+        public string DateTimeInfo { get; set; }
+
+        [DataMember(Name = "m_dataItem", IsRequired = true)]
+        public int DataItem
+        {
+            get { return _dataItem; }
+            set { _dataItem = value; }
+        }
+
+        [DataMember(Name = "m_isReadOnly", IsRequired = true)]
+        public bool IsReadOnly
+        {
+            get { return _isReadOnly; }
+            set { _isReadOnly = value; }
+        }
+
+        [DataMember(Name = "m_name", IsRequired = true)]
+        public string Name
+        {
+            get { return _name; }
+            set { _name = value; }
+        }
+
+        [DataMember(Name = "m_useUserOverride", IsRequired = true)]
+        public bool UseUserOverride
+        {
+            get { return _useUserOverride; }
+            set { _useUserOverride = value; }
+        }
+
+        [DataMember(Name = "numInfo", IsRequired = true)]
+        public string NumInfo { get; set; }
+
+        [DataMember(Name = "textInfo", IsRequired = true)]
+        public string TextInfo { get; set; }
+
+        public static CultureInfo GetCultureInfo(CultureInfoAdapter value)
+        {
+            return new CultureInfo(value.Name);
+        }
+
+        public static CultureInfoAdapter GetCultureInfoAdapter(CultureInfo value)
+        {
+            return new CultureInfoAdapter()
+            {
+                Calendar = null,
+                CompareInfo = null,
+                CultureID = 2052,
```

Comment:
This field does not exist on CoreCLR. This is the default value on Desktop. I added this just to make the payload compatible with Desktop's.

I just found that this field is marked with `OptionalFieldAttribute` on Desktop, which means it can be missing from a serialization stream. I'll remove the field.